lib/dao.js

```diff
@@ -2687,6 +2687,14 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
   Model.notifyObserversOf('before save', context, function(err, ctx) {
     if (err) return cb(err);
+    if (Model.settings.forceId && ctx.instance[pkName] !== id) {
+      var err = new Error ('id property cannot be changed from ' +
+        id + ' to ' + inst[pkName] + ' in \'before save\' operation ' +
+        'hook when forceId is set to true');
```

Comment:
Is it a good idea to allow hooks to change the id even when `forceId` is not set? Can the connectors handle this?

Regardless of the decision, could you please add a unit-test to demonstrate & verify what happens when a hook modifies the id property?